core_new/src/message/common.rs

```diff
@@ -0,0 +1,163 @@
+// This file is part of Gear.
+
+// Copyright (C) 2022 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::identifiers::{MessageId, ProgramId};
+use crate::message::{
+    DispatchKind, ExitCode, GasLimit, Payload, StoredDispatch, StoredMessage, Value,
+};
+use codec::{Decode, Encode};
+use core::ops::Deref;
+use scale_info::TypeInfo;
+
+/// Message.
+#[derive(Clone, Default, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, Decode, Encode, TypeInfo)]
+pub struct Message {
+    /// Message id.
+    id: MessageId,
+    /// Message source.
+    source: ProgramId,
+    /// Message destination.
+    destination: ProgramId,
+    /// Message payload.
+    payload: Payload,
+    /// Message optional gas limit.
+    gas_limit: Option<GasLimit>,
+    /// Message value.
+    value: Value,
+    /// Message id replied on with exit code.
+    reply: Option<(MessageId, ExitCode)>,
+}
+
+impl Message {
+    /// Create new message.
+    pub fn new(
```

Comment:
This constructor is not quite useful as we can replace is by 

```rust
Message {
    id: ...,
    source: ...,
    destination: ...,
    etc.
}
```

Reply:
> This constructor is not quite useful as we can replace is by
> 
> ```rust
> Message {
>     id: ...,
>     source: ...,
>     destination: ...,
>     etc.
> }
> ```

This constructor need all fields public